tlslite/tlsconnection.py

```diff
@@ -424,6 +425,9 @@ def _handshakeClientAsyncHelper(self, srpParams, certParams, anonParams,
                 raise ValueError("Caller passed no nextProtos")
         if alpn is not None and not alpn:
             raise ValueError("Caller passed empty alpn list")
+        if serverName and not is_valid_hostname(serverName):
```

Comment:
Thanks for clarification. Maybe mentioning this in the comment would be useful, since it can be confusing for further readers?